### PR TITLE
Add error messages to the Shop

### PR DESCRIPTION
**Pull request by amiskov**
## What issue does this PR close
Part of #419.

## Changes Proposed ( a list of new changes introduced by this PR)
- [x] Add better error messages.
- [x] Redirect the user to the main Shop page only if the transaction was successful.
- [x] Get rid of "Impossible error" after buying a product.

## How to test ( a list of instructions on how to test this PR)
Try to buy something in the shop. Make sure to check some error states, like buying 0 products.

### CODE REVIEW

**Review by lucca65 — APPROVED** [review]
so we weren't even handling the blockchain callback.... 😢 

Thanks for that @amiskov

File: public/translations/cat-CAT.json [hunk 1]
```diff
@@ -99,7 +99,18 @@
...
+        "Can't sale for yourself": "No es pot vendre per vosaltres mateixos",
+        "Can't find any sale with given sale_id": "No trobo cap venda amb sale_id determinat",
+        "Invalid number of units, must be greater than 0": "El nombre d'unitats no vàlid ha de ser superior a 0",
+        "Sale doesn't have that many units available": "La venda no té tantes unitats disponibles",
+        "Amount offered doesn't correspond to expected value": "L'import ofert no correspon al valor esperat",
+        "Quantity must be the same as the sale price": "La quantitat ha de ser la mateixa que el preu de venda",
+        "Can't find community with given Symbol": "No es pot trobar la comunitat amb el símbol donat",
+        "This community don't have objectives enabled.": "Aquesta comunitat no té objectius activats.",
+        "You can't use transfersale to this sale if you aren't part of the community": "No podeu fer servir la venda de transferències a aquesta venda si no formeu part de la comunitat",
+        "Sale creator and sale doesn't match": "El creador de la venda i la venda no coincideixen"
```
Comment (lucca65) [inline public/translations/cat-CAT.json:113]: Actually, can you add one more error here? I'll keep the same string description on the contract when we implement it, but its

`Invalid number of units`

**Review by lucca65 — CHANGES_REQUESTED** [review]

File: public/translations/cat-CAT.json (hunk 1 above)
Comment (amiskov) [reply public/translations/cat-CAT.json]: @lucca65 done!

File: public/translations/en-US.json [hunk 2]
```diff
@@ -95,11 +95,23 @@
         "destination account doesn't exists": "Destination account doesn't exist",
         "token with given symbol doesn't exists": "Token with given symbol doesn't exist",
         "invalid quantity": "Invalid quantity",
-        "quantity must be positive": "La quantitat ha de ser positiva",
+        "quantity must be positive": "Quantity must be positive",
```
Comment (lucca65) [inline public/translations/en-US.json:98]: nice catch

**Review by lucca65 — APPROVED** [review]

**Pull request merged**